lib/plugins/aws/package/compile/functions/index.js

```diff
@@ -66,6 +66,34 @@ class AwsCompileFunctions {
     }
   }
 
+  downloadS3Artifact(functionName) {
```

Comment:
IMHO this function name should be different since we're only downloading it when the package is an S3 URL.